### ISSUE TITLE
Breakpoints not hit in debug mode on intellij

### ISSUE THREAD

**Issue by scottiedog45** [GoogleCloudPlatform/cloud-code-intellij#3122]
(Please ensure you are running the latest version of Cloud Cloud IntelliJ with _Help > Check for Updates_.)

(screenshots are often helpful)

### Expected Behavior

Normal breakpoint behavior when running debug mode, like pausing at breakpoints. 

### Actual Behavior

Breakpoints skipped/not hit

### Additional Information

Trying to run the guestbook application.

```[backend] 2023-01-08T04:27:01Z error layer=rpc writing response:write tcp 127.0.0.1:56268->127.0.0.1:41584: use of closed network connection
[frontend] 2023-01-08T04:27:01Z error layer=rpc writing response:write tcp 127.0.0.1:56268->127.0.0.1:41570: use of closed network connection```

I am able to hit breakpoints when running the project through VS Code. 

- IDE type (e.g. IntelliJ, Pycharm): Intellij
- IDE version: 2022.2.5
- Cloud Code version (Settings > Plugins > Cloud Code): 22-12-2.222
- Cloud SDK (Settings > Cloud Code > Cloud SDK)
  - Are you allowing the plugin to manage the Cloud SDK: Yes
  - Version of the Cloud SDK: 412.0.0
- Skaffold version (If you are using Kubernetes features; Settings > Cloud Code > Kubernetes): skaffold v2.0.4
- Operating System: 13.1


_Feel free to deviate from this template as needed, especially if you are submitting a feature request._

**Comment by etanshaul** [comment]
hi @scottiedog45 - I see you are using a Go application. Did you enable `go modules` support for your project? This [is required](https://cloud.google.com/code/docs/intellij/debug#go) for debugging Go containers with Cloud Code IntelliJ.

From that link:

> To configure your application for debugging, your app must be a [Go Module](https://blog.golang.org/using-go-modules)-based application and be identified as being Go-based by setting one of the [standard Go runtime environment variables](https://godoc.org/runtime) in the container, such as GODEBUG, GOGC, GOMAXPROCS, or GOTRACEBACK. GOTRACEBACK=single is the default setting for Go and GOTRACEBACK=all is a generally useful configuration.

To enable Go modules, go to Settings > Languages & Frameworks > Go > Go Modules, and select `Enable Go modules integration`:
<img width="416" alt="Screenshot 2023-01-09 at 10 01 23 AM" src="https://user-images.githubusercontent.com/1735744/211338811-859a2406-5a48-41f3-905a-e654f836fa2a.png">

**Comment by scottiedog45** [comment]
Hey @etanshaul , thanks for the speedy reply. 

Can confirm modules are enabled: 

![Screenshot 2023-01-09 at 10 16 23 AM](https://user-images.githubusercontent.com/31543854/211342276-a78b83f0-a129-490b-bd83-164bca2c65f2.png)

Also on Go version 1.19.4 if that helps.

**Comment by etanshaul** [comment]
I see. I am not able to reproduce with the go-guestbook application.

Are you able to see the debug tabs as shown here:
<img width="754" alt="Screenshot 2023-01-09 at 10 28 18 AM" src="https://user-images.githubusercontent.com/1735744/211344952-203fe254-a7dc-4269-9099-6ea7858892c2.png">

Also, just in case (I've made this mistake before) - where did you set your breakpoint? You won't be able to debug the server start up code as-is (`main()`) as the debugger is attached later. Instead, try setting a breakpoint in the `homeHandler`:
<img width="918" alt="Screenshot 2023-01-09 at 10 30 18 AM" src="https://user-images.githubusercontent.com/1735744/211345217-bc2fead8-e956-41ad-8790-7a37db382564.png">

Otherwise, is there anything you may have changed in your application?

**Comment by scottiedog45** [comment]
Yep I can see the debug tabs, but there is no output when the application starts:

![image](https://user-images.githubusercontent.com/31543854/211347094-c4216296-c578-4091-b933-2eb754954667.png)

I made a fresh project, just to be sure I haven't changed anything. Still seeing the same output. Not sure if these lines from the terminal output are suspicious:

`Some taggers failed. Rerun with -vdebug for errors.`
```
[frontend] 2023-01-09T15:37:34Z error layer=rpc writing response:write tcp 127.0.0.1:56268->127.0.0.1:34416: use of closed network connection
[backend] 2023-01-09T15:37:34Z error layer=rpc writing response:write tcp 127.0.0.1:56268->127.0.0.1:34418: use of closed network connection
```

**Comment by etanshaul** [comment]
fwiw, the `console` tab is also blank for me, and I also see the following messages in the verbose logs:

> [frontend] 2023-01-09T16:39:31Z error layer=rpc writing response:write tcp 127.0.0.1:56268->127.0.0.1:51758: use of closed network connection
> [backend] 2023-01-09T16:39:31Z error layer=rpc writing response:write tcp 127.0.0.1:56268->127.0.0.1:48174: use of closed network connection

Yet, I can still hit the breakpoints.

You should be able to get a bunch of additional debug logs if you enable verbose debugging on Skaffold. To do this, in the Run Configuration, under `advanced settings`, set `logging verbosity` to `DEBUG`:
<img width="771" alt="Screenshot 2023-01-09 at 11 42 32 AM" src="https://user-images.githubusercontent.com/1735744/211360986-4ef2be91-685e-4204-b8e0-f66bba54bcb7.png">

If you search for the keyword "debug" in the log, you should see several bits on information. For example this snippet showing successful application of the debug transformations:

```
Configuring "" for Go/Delve debugging
"frontend" requires debugging support image "go"
Configuring installation of debugging support files
Applied debugging transform:
 apiVersion: apps/v1
kind: Deployment
```

**Comment by etanshaul** [comment]
here is a gist containing my debug level logs for the go-guestbook app (with working breakpoint debugging), in case it helps for comparison.: https://gist.github.com/etanshaul/4d27a3244e17b0ad68d16d9f143e0553

If you share yours, I can take a look too

**Comment by scottiedog45** [comment]
Thanks again for your help with this, and easing my mind about those logs before. 

Here is a [link](https://gist.github.com/scottiedog45/c185fb209594ca731fb567c864a7131e) to my gist of logs. 

One difference I can see is regarding tags. Yours says 

```
Tags used in deployment:
 - go-guestbook-frontend -> gcr.io/etan-gcp/go-guestbook-frontend:v2.0.1-1-gaa0bf348b-dirty@sha256:4186360accc42b675ca32aeb216d4a78881d2486e0d929d1b7d608115cc89f60
 - go-guestbook-backend -> gcr.io/etan-gcp/go-guestbook-backend:v2.0.1-1-gaa0bf348b-dirty@sha256:e9b59ccc2d56c0a8532175e888df09a23aacadb3ecdd4dbfed365d2f46acdd5c
push value not present in isImageLocal(), defaulting to true because cluster.PushImages is true
push value not present in isImageLocal(), defaulting to true because cluster.PushImages is true
```

while mine is 

```
Tags used in deployment:
 - go-guestbook-frontend -> go-guestbook-frontend:3d6a7afe103cef8afbc064da54b654c69ceabbac8d1ee73d53b94038e94f8e35
 - go-guestbook-backend -> go-guestbook-backend:09130c2d49256a3c65c898d5fcdd90315874ea2a009c9b33ad1c8d1f93aa7c99
push value not present in isImageLocal(), defaulting to false because cluster.PushImages is false
push value not present in isImageLocal(), defaulting to false because cluster.PushImages is false
Local images can't be referenced by digest.
They are tagged and referenced by a unique, local only, tag instead.
See https://skaffold.dev/docs/pipeline-stages/taggers/#how-tagging-works
```

Here is a copy of my rc, to verify that I'm trying to deploy to a local cluster

![image](https://user-images.githubusercontent.com/31543854/211418161-a8cc7de0-680e-4214-9f02-f6acd395e1b0.png)

**Comment by etanshaul** [comment]
strange. The only other difference I noticed here is that you are deploying to a minikube cluster (whereas I was using GKE). I tried minikube however, and it worked fine, so this is a bit perplexing.

I will give the logs another closer look and see if we have any ideas. 

In the meantime - could you try the simple hello-world Go app (not the guestbook app)? It is in the New Project wizard as well. I wonder if it is that app specifically.

**Comment by scottiedog45** [comment]
I can hit breakpoints in the hello-world app 🥳 .

I tried the guest book app after again, and noticed that several debugger tabs open instead of two (i.e. one for the backend and frontend). This time, two were opened for the front end? Also, the breakpoints shift to an "invalid" state when the front end tabs are selected, even though the breakpoints are in the frontend:

![image](https://user-images.githubusercontent.com/31543854/211867747-804a932c-8236-4764-a435-fa2c5de9f73f.png)
